Empire - The Last Battle/Assets/Unity/Scripts/MonoBehaviours/OverworldManager.cs

```diff
@@ -38,12 +38,80 @@ void _OverworldUI_OnCommanderMove(TileData tile)
 
         //****JUST FOR TESTING**** set new reachable tiles
         _OverworldUI.AllowPlayerMovement(_Board.GetReachableTiles(_BattlebeardPlayer.CommanderPosition, 1));
+
+        //Handles events that happen when player lands on that tile
+        HandleTileEvent(tile);
+    }
+
+    void HandleTileEvent(TileData tile)
+    {
+        switch (tile.Building)
+        {
+            case BuildingType.None:
+                break;
+            case BuildingType.Armoury:
+                break;
+            case BuildingType.Camp:
+                break;
+            case BuildingType.CastleBattlebeard:
+                break;
+            case BuildingType.CastleStormshaper:
+                break;
+            case BuildingType.Cave:
+                CaveGiveCard();
+                break;
+            case BuildingType.Fortress:
+                break;
+            case BuildingType.Inn:
+                break;
+            case BuildingType.StartTileBattlebeard:
+                break;
+            case BuildingType.StartTileStormshaper:
+                break;
+            default:
+                break;
+        }
+
+        switch (tile.Terrain)
+        {
+            case TerrainType.CastleCorner00:
+                break;
+            case TerrainType.CastleCorner01:
+                break;
+            case TerrainType.CastleCorner10:
+                break;
+            case TerrainType.CastleCorner11:
+                break;
+            case TerrainType.Forest:
+                break;
+            case TerrainType.Grass:
+                break;
+            case TerrainType.Mountain:
+                break;
+            case TerrainType.Swamp:
+                break;
+            case TerrainType.Tundra:
+                break;
+            default:
+                break;
+        }
+    }
+
+    public void CaveGiveCard()
```

Comment:
Maybe make this a bit more generic? Accepting a list of cards and returning one of them